src/Features/Core/Portable/UseSystemHashCode/UseSystemHashCodeDiagnosticAnalyzer.cs

```diff
@@ -49,14 +49,10 @@ private void AnalyzeOperationBlock(Analyzer analyzer, OperationBlockAnalysisCont
                 return;
             }
 
+            var syntaxTree = operation.Syntax.SyntaxTree;
             var cancellationToken = context.CancellationToken;
-            var optionSet = context.Options.GetDocumentOptionSetAsync(operation.Syntax.SyntaxTree, cancellationToken).GetAwaiter().GetResult();
-            if (optionSet == null)
-            {
-                return;
-            }
 
-            var option = optionSet.GetOption(CodeStyleOptions.PreferSystemHashCode, operation.Language);
+            var option = context.Options.GetOption(CodeStyleOptions.PreferSystemHashCode, operation.Language, syntaxTree, cancellationToken);
```

Comment:
No fancy extension method for this one? Fine if there isn't, just not sure if this got missed.

Reply:
No fancy one for this one. Not a lot of Operation Block Analyzers in Roslyn.